realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ProcessCommitTests.java

```diff
@@ -16,159 +16,202 @@
 
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
-import java.util.concurrent.TimeUnit;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
 
-import io.realm.ObjectServerError;
 import io.realm.Realm;
 import io.realm.RealmChangeListener;
 import io.realm.RealmResults;
 import io.realm.SyncConfiguration;
-import io.realm.SyncSession;
 import io.realm.SyncUser;
 import io.realm.objectserver.model.ProcessInfo;
 import io.realm.objectserver.model.TestObject;
-import io.realm.objectserver.service.SendOneCommit;
-import io.realm.objectserver.service.SendsALot;
 import io.realm.objectserver.utils.Constants;
+import io.realm.objectserver.utils.RemoteIntegrationTestService;
 import io.realm.objectserver.utils.UserFactory;
+import io.realm.rule.RunInLooperThread;
+import io.realm.rule.RunTestInLooperThread;
+import io.realm.rule.RunTestWithRemoteService;
+import io.realm.rule.RunWithRemoteService;
+import io.realm.rule.TestSyncConfigurationFactory;
+import io.realm.services.RemoteTestService;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
 
 
 @RunWith(AndroidJUnit4.class)
 public class ProcessCommitTests extends BaseIntegrationTest {
 
+    @Rule
+    public RunInLooperThread looperThread = new RunInLooperThread();
+    @Rule
+    public RunWithRemoteService remoteService = new RunWithRemoteService();
+    @Rule
+    public TestSyncConfigurationFactory configFactory = new TestSyncConfigurationFactory();
+
+    @Before
+    public void before() throws Exception {
+        UserFactory.resetInstance();
+    }
+
+    public static class SimpleCommitRemoteService extends RemoteIntegrationTestService {
+        private static SyncUser user;
+        public static final Step stepA_openRealmAndCreateOneObject = new Step(RemoteTestService.BASE_SIMPLE_COMMIT, 1) {
+
+            @Override
+            protected void run() {
+                user = UserFactory.getInstance().loginWithDefaultUser(Constants.AUTH_URL);
+                String realmUrl = Constants.SYNC_SERVER_URL;
+
+                final SyncConfiguration syncConfig = new SyncConfiguration.Builder(user, realmUrl)
+                        .directory(getService().getRoot())
+                        .build();
+                getService().realm = Realm.getInstance(syncConfig);
+                Realm realm = getService().realm;
+
+                realm.beginTransaction();
+                ProcessInfo processInfo = realm.createObject(ProcessInfo.class);
+                processInfo.setName("Background_Process1");
+                processInfo.setPid(android.os.Process.myPid());
+                processInfo.setThreadId(Thread.currentThread().getId());
+                realm.commitTransaction();
+                // FIXME: If we close the Realm here, the data won't be able to synced to the main process. Is it a bug
```

Comment:
Not currently, no. This is also something we need to get fixed, because other features depend on this being possible